sympy/polys/series/flint_powerseriesring.py

```diff
@@ -0,0 +1,819 @@
+            return s * n
+
+    def divide(self, s1: ZZSeries, s2: ZZSeries) -> ZZSeries:
+        """
+        Divide two power series.
+        """
+        if isinstance(s1, fmpz_poly):
+            s1 = fmpz_series(s1.coeffs(), prec=self._prec)
+        if isinstance(s2, fmpz_poly):
+            s2 = fmpz_series(s2.coeffs(), prec=self._prec)
+
+        with _global_cap(self._prec):
+            return s1 / s2
+
+    def pow_int(self, s: ZZSeries, n: int) -> ZZSeries:
+        """
+        Raise a power series to an integer power.
+        """
+        if n < 0:
+            raise ValueError("Power must be non-negative")
+
+        if isinstance(s, fmpz_poly):
+            poly = s**n
+            if poly.degree() < self._prec:
+                return poly
+            return fmpz_series(poly, prec=self._prec)
+
+        with _global_cap(self._prec):
+            return s**n
+
+    def square(self, s: ZZSeries) -> ZZSeries:
+        """
+        Return the square of a power series.
+        """
+        return self.pow_int(s, 2)
+
+    def compose(self, s1: ZZSeries, s2: ZZSeries) -> ZZSeries:
+        """
+        Compose two power series.
+        """
+        dom = self._domain
+        ring_prec = self._prec
+        dup1 = dup_reverse(s1.coeffs())
+        dup2 = dup_reverse(s2.coeffs())
+
+        if isinstance(s1, fmpz_poly) and isinstance(s2, fmpz_poly):
+            comp = dup_series_compose(dup1, dup2, ring_prec, dom)
+            comp = comp[::-1]
+
+            deg1 = s1.degree()
+            deg2 = s2.degree()
+            if deg1 + deg2 < ring_prec:
+                return fmpz_poly(comp)
+            else:
+                return fmpz_series(comp, prec=ring_prec)
+
+        if dup2 and not dom.is_zero(dup2[-1]):
+            raise ValueError(
+                "Series composition requires the second series to have a zero constant term."
+            )
+
+        prec1 = _get_series_precision(s1) if isinstance(s1, fmpz_series) else ring_prec
+        prec2 = _get_series_precision(s2) if isinstance(s2, fmpz_series) else ring_prec
+
+        DUP1 = (dup1, prec1)
+        DUP2 = (dup2, prec2)
+
+        coeffs1, coeffs2, min_prec = _unify_prec(DUP1, DUP2, dom, ring_prec)
+        comp = dup_series_compose(coeffs1, coeffs2, min_prec, dom)
+        comp = comp[::-1]
+        return fmpz_series(comp, prec=min_prec)
+
+    def inverse(self, s: ZZSeries) -> ZZSeries:
+        """
+        Compute the inverse of a power series.
+        """
+        dom = self._domain
+        coeffs = s.coeffs()
+
+        if not coeffs or not dom.is_unit(coeffs[0]):
+            raise NotReversible(
+                "Series inverse requires the constant term to be a unit"
+            )
+
+        prec = _get_series_precision(s) if isinstance(s, fmpz_series) else self._prec
+
+        dup = dup_reverse(coeffs)
+        inv = dup_revert(dup, prec, dom)
+        inv = dup_truncate(inv, prec, dom)[::-1]
+        return fmpz_series(inv, prec=prec)
```

Comment:
I think it is clearer if you always handle reversing the list at the boundary either when calling coeffs or when calling fmpz_series:
```python
coeffs = s.coeffs()[::-1]
...
return fmpz_series(inv[::-1], prec=prec)
```
Then it is clear what the order is for all of the rest of the code.